include/api.php

```diff
@@ -3664,6 +3667,336 @@ function api_friendica_notification_seen($type){
+		if($r == null) {
+			$answer = array('result' => 'error', 'message' => 'no mails available');
+			return api_format_data("direct_messages_all", $type, array('$result' => $answer));
+		}
+
+		$ret = Array();
+		foreach($r as $item) {
+			if ($box == "inbox" || $item['from-url'] != $profile_url){
+				$recipient = $user_info;
+				$sender = api_get_user($a,normalise_link($item['contact-url']));
+			}
+			elseif ($box == "sentbox" || $item['from-url'] == $profile_url){
+				$recipient = api_get_user($a,normalise_link($item['contact-url']));
+				$sender = $user_info;
+
+			}
+			$ret[]=api_format_messages($item, $recipient, $sender);
+		}
+
+
+		$data = array('$messages' => $ret);
+		switch($type){
+			case "atom":
+			case "rss":
+				$data = api_rss_extra($a, $data, $user_info);
+		}
+
+		return  api_format_data("direct_messages_all", $type, $data);
+
+	}
+	api_register_func('api/friendica/direct_messages_all', 'api_friendica_direct_messages_all', true);
+
+
+	/**
+	 * @brief update a direct_message to seen state for Windows 10 App 
+	 *
+	 * @param App $a
+	 * @param string $type Known types are 'atom', 'rss', 'xml' and 'json'
+	 * @return string
+	 */
+	function api_friendica_direct_messages_setseen($type){
+		$a = get_app();
+		if (api_user()===false) throw new ForbiddenException();
+
+		// params
+		$user_info = api_get_user($a);
+		$uid = $user_info['uid'];
+		$id = (x($_REQUEST, 'id') ? $_REQUEST['id'] : 0);
+
+		// return error if id is zero
+		if ($id == "") {
+			$answer = array('result' => 'error', 'message' => 'message id not specified');
+			return api_format_data("direct_messages_setseen", $type, array('$result' => $answer));
+		}
+
+		// get data of the specified message id
+		$r = q("SELECT * FROM `mail` WHERE `id` = %d AND `uid` = %d",
+			intval($id), 
+			intval($uid));
+		// error message if specified id is not in database
+		if (count($r) == 0) {
+			$answer = array('result' => 'error', 'message' => 'message id not in database');
+			return api_format_data("direct_messages_setseen", $type, array('$result' => $answer));
+		}
+
+		// update seen indicator
+		$result = q("UPDATE `mail` SET `seen` = 1 WHERE `id` = %d AND `uid` = %d", 
+			intval($id), 
+			intval($uid));
+
+		if ($result) {
+			// return success
+			$answer = array('result' => 'ok', 'message' => 'message set to seen');
+			return api_format_data("direct_message_setseen", $type, array('$result' => $answer));
+		} else {
+			$answer = array('result' => 'error', 'message' => 'unknown error');
+			return api_format_data("direct_messages_setseen", $type, array('$result' => $answer));
+		}
+	}
+	api_register_func('api/friendica/direct_messages_setseen', 'api_friendica_direct_messages_setseen', true);
+
+
+	/**
+	 * @brief delete a direct_message from mail table through api
+	 *
+	 * @param App $a
+	 * @param string $type Known types are 'atom', 'rss', 'xml' and 'json'
+	 * @return string
+	 */
+	function api_friendica_direct_messages_delete($type){
```

Comment:
Could you implement the Twitter standard API call for this functionality instead of creating something new?
https://dev.twitter.com/rest/reference/post/direct_messages/destroy